### DISCUSSION TITLE
Converting deposit to transfer

### DISCUSSION THREAD

**Discussion by Snyaify** [firefly-iii/firefly-iii#5869] [Firefly III]
Good day
After 3 hours of trying to convert a deposit into a transfer, I want to ask for help, because with what I encountered it looks like a bug

I have a default asset account with mark of my card *2711 and i'm doing import with data importer on this default account
when I created a rule that converts a deposit into a transfer (see screenshots) This transaction is created as a transfer, but for some reason it does not go to the main account (Asset account), but is created in the section Revenue accounts.
(I'm trying to convert deposit transaction into transfer because i have  2 different accounts within 1 bank)
Main account
Family account
Trying to convert deposit from main to transfer on Family
![browser_5Nosrud1sC](https://user-images.githubusercontent.com/55499522/157448683-c03e6e57-9412-4fc3-8277-387f2516b4f1.jpg)
![browser_22BhxFQdqS](https://user-images.githubusercontent.com/55499522/157448739-e3e9931a-9cd8-498e-8a28-746f239e0bb6.jpg)
![browser_5pjvwjVhtX](https://user-images.githubusercontent.com/55499522/157448780-4aa94a48-2952-4c30-80d4-c1b40cf8e797.jpg)
![browser_IeM5ftSDTF](https://user-images.githubusercontent.com/55499522/157448979-9232cc54-10a9-4dbd-9bee-ee41cf4d3d3d.jpg)
![browser_wWw4Aj9HVw](https://user-images.githubusercontent.com/55499522/157449193-46990a8e-6021-49c1-9e7a-af45eeb3a20a.jpg)

**Comment by JC5** [comment]
That's a bug, thanks for letting me know. Are you using Docker?

**Reply by Snyaify** [reply]
Yes, docker on unraid

**Reply by JC5** [reply]
Try the latest `:develop` tag, that should fix the issue.

**Reply by Snyaify** [reply]
I tried, but unfortunately it didn't help.
Maybe I didn’t express myself correctly initially, I tried to record a video of the process of importing the operation, maybe this will clarify the situation with my problem
[https://tmpfiles.org/234252/browser_webehc2sgb.mp4](https://tmpfiles.org/234252/browser_webehc2sgb.mp4)

**Reply by JC5** [reply]
Sorry, I'm getting a 404.

**Reply by Snyaify** [reply]
Sorry, tmpfiles removed file after 3h, here is [https://dropmefiles.com/TwyhR](https://dropmefiles.com/TwyhR)

**Reply by JC5** [reply]
Another 404, I'm afraid :(

**Reply by Snyaify** [reply]
I hope this will work :\
[google drive link](https://drive.google.com/file/d/1RmFofh_J3Y1ud5pbtooV0LOKFtYs_xTD/view?usp=sharing)

**Reply by JC5** [reply]
I got it!

**Reply by Snyaify** [reply]
Any news on my issue? :)

**Reply by JC5** [reply]
I have not been able to reproduce the issue on the latest develop version.

- A deposit is imported. The source is a revenue account, the destination an asset account.
- Converting it to a transfer works, and the source account is replaced by an existing asset account

The asset account has to exist for the rule to work.

- Changing the destination account also works, if it's an asset account. If the asset account does not exist, the rule silently fails.

**Comment by PiotrEsse** [comment]
Sorry for reviving old post but I think its better to ask here than create new post. 
After setting automatic rule for converting transaction into transfer I am getting additional transfer transaction but original transaction remains. Its ok?

**Reply by JC5** [reply]
Please open a new post, we won't run out.

**Comment by KaosLtd** [comment]
I've encountered the same issue (I think).

1. Used the importer to import transactions.
2. Categorised all transactions with type 'TRF' on my statement as 'Transfer'
3. All imported transactions that were transfers INTO my current account were imported as 'deposits', with source as '(no name)' and destination as 'current' (asset account)
4. Created a rule to find all transactions that are categorised as 'transfer' but are NOT 'transfers' (type).
5. Running the rule's filter successfully lists all the transactions I want.
6. Added an action to convert all matching transactions to 'transfers' with Action Value of 'savings' (asset account)
7. When I run the rule, it says nothing was converted.

I have manually converted one of these successfully, but I'd rather not manually convert ~300 transactions individually.

I had previously successfully run a rule to convert all imported actions that were transfers from 'savings' to 'current' (I assume these transactions were imported as 'withdrawals'?).
